var/spack/repos/builtin/packages/circuit-build/package.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class CircuitBuild(PythonPackage):
+    """Command Line API for building circuits"""
+
+    homepage = "https://bbpcode.epfl.ch/code/#/admin/projects/common/circuit-build"
+    git      = "ssh://bbpcode.epfl.ch/common/circuit-build"
+
+    version('develop', branch='master')
+    version('3.1.1', tag='circuit-build-v3.1.1')
+
+    depends_on('py-setuptools', type=('build', 'run'))
+
+    depends_on('py-click@7.0:7.99', type='run')
+    depends_on('snakemake@5.6:5.99', type='run')
```

Comment:
@matz-e can you please advice. This 'snakemake@5.6:5.99' dependency provides an executable command `snakemake` which does not become available after the module `circuit-build` is loaded. Should I add 'snakemake' to `deploy/configs/applications/modules.yaml` and `deploy/packages/bbp-packages.yaml` in order to provide `snakemake` executable after the module of 'circuit-build' has been loaded?
```
@bbpv1 ~$ module load unstable circuit-build
Autoloading python/3.7.4
@bbpv1 ~$ pip list
Package             Version  
------------------- ---------
...
snakemake           5.10.0   
...
@bbpv1 ~$ snakemake
command not found: snakemake
```

Reply:
There's already a snakemake module, don't do that.

You can explicitly request `snakemake`'s `bin` in the path by adding something like
```
def setup_run_environment(self, env):
    env.prepend_path('PATH', self.spec['snakemake'].prefix.bin)
```
grep through the sources to find examples about the exact syntax